src/main/java/io/github/oliviercailloux/y2018/apartments/valuefunction/profile/QuestionPriceArea.java

```diff
@@ -0,0 +1,109 @@
+package io.github.oliviercailloux.y2018.apartments.valuefunction.profile;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+import com.google.common.collect.Range;
+import io.github.oliviercailloux.y2018.apartments.valuefunction.Criterion;
+
+public class QuestionPriceArea {
+  int price;
+  int surface;
+  private final String question = "Would you pay %d€ more for %d m2 more?";
+
+  /**
+   * Instantiates a new question price/area.
+   *
+   * @param price the price shown to the user according to his/her profile
+   * @param surface the surface shown to the user according to his/her profile
+   */
+  private QuestionPriceArea(int price, int surface) {
+    checkNotNull(price);
+    checkNotNull(surface);
+    this.price = price;
+    this.surface = surface;
+  }
+
+  /**
+   * Creates the object question.
+   *
+   * @param price the price
+   * @param surface the surface
+   * @return the question price area
+   */
+  public static QuestionPriceArea create(int price, int surface) {
+    return new QuestionPriceArea(price, surface);
+  }
+
+  /**
+   * Getter for attribute price
+   *
+   * @return the price : question parameter
+   */
+  public int getPrice() {
+    return this.price;
+  }
+
+  /**
+   * Getter for attribute surface
+   *
+   * @return the surface : question parameter
+   */
+  public int getSurface() {
+    return this.surface;
+  }
+
+  /**
+   * Build the question according to parameters emitted during class creation
+   *
+   * @return the question constructed
+   */
+  public String getQuestion() {
+    return String.format(this.question, this.getPrice(), this.getSurface());
+  }
+
+  /**
+   * Adapt the interval of weight contained in a profile depending on his response
+   *
+   * @param p the profile we need to modify
+   * @param response the response of the user
+   */
+  public void resolve(Profile p, boolean response) {
+    checkNotNull(p);
+    checkNotNull(response);
```

Comment:
Idem que précédemment, un boolean ne peut pas être null il me semble (ce check serait pertinent si l'on avait un Boolean à la place d'un boolean).

Reply:
Fix in 5cb20dc6dd6f860db028a2a910b1575f02bdd942
Pour information, un boolean `null` prend automatiquement la valeur `false`